src/wallet/rpcwallet.cpp

```diff
@@ -88,12 +89,50 @@ namespace{
     {
         int shieldedPoolDeprecationForkHeight = GetShieldedPoolDeprecationForkHeight();
 
-        return static_cast<std::string>("\nWARNING: This method ") +
-                                       ((chainActive.Height() + 1 >= shieldedPoolDeprecationForkHeight) ? "has been " : "is going to be ") +
+        return static_cast<std::string>("\nWARNING: This method has been ") +
                                        (fullDeprecation ? "fully " : "partially ") +
                                         "deprecated at block height " + std::to_string(shieldedPoolDeprecationForkHeight - 1) +
                                         " due to shielded pool deprecation hard fork.";
     }
+
+
+    const char* unshieldingToScriptOnlyErrorMessage = "This method is deprecated due to unshielding to script only hard fork.";
+
+    //! Method for getting unshielding to script only hard fork height
+    /*!
+      \return hard fork height
+    */
+    int GetUnshieldingToScriptOnlyForkHeight()
+    {
+        CBaseChainParams::Network network = Params().NetworkIDString() == "main" ? CBaseChainParams::Network::MAIN :
+                                            Params().NetworkIDString() == "test" ? CBaseChainParams::Network::TESTNET :
+                                                                                   CBaseChainParams::Network::REGTEST;
+        return UnshieldingToScriptOnlyFork{}.getHeight(network);
+    }
+
+    //! Method for identifying if unshielding RPC methods are deprecated
+    /*!
+      \return shielding RPC methods deprecated
+    */
+    bool AreUnshieldingRPCMethodsDeprecated()
+    {
+        return chainActive.Height() + 1 >= GetUnshieldingToScriptOnlyForkHeight();
+    }
```

Comment:
Addressed all instances except for `std::string UnshieldingRPCMethodsDeprecationWarning()` (in order to avoid calling twice `GetUnshieldingToScriptOnlyForkHeight`).